src/test.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState } from "react"
+
+import "@testing-library/jest-dom/extend-expect"
+
+import { render, fireEvent } from "@testing-library/react"
+import { useKeyboardShortcuts } from "."
+
+const TestComponent = () => {
+  const [key, setKey] = useState<string>("Nothing pressed")
+
+  useKeyboardShortcuts([
+    { keys: ["a"], onEvent: () => setKey("Pressed a") },
+    { keys: ["a"], onEvent: () => setKey("Pressed a") },
+    { keys: ["Shift", "Minus"], onEvent: () => setKey("Pressed ?") },
+    { keys: ["9"], onEvent: () => setKey("Pressed 9") },
+    { keys: ["ctrl", "a"], onEvent: () => setKey("Pressed ctrl + a") },
+    {
+      keys: ["ctrl", "shift", "a"],
+      onEvent: () => setKey("Pressed ctrl + shift + a"),
+    },
+    {
+      keys: ["ctrl", "shift", "alt", "a"],
+      onEvent: () => setKey("Pressed ctrl + shift + alt + a"),
+    },
+  ])
+
+  useKeyboardShortcuts(
+    [
+      {
+        keys: ["ctrl", "shift"],
+        onEvent: () => setKey("Scrolled + ctrl + shift"),
+      },
+    ],
+    true,
+    [],
+    "wheel"
+  )
+  return <>{key}</>
+}
+
+describe("useKeyboardShortcuts", () => {
+  it("handles single characters", async () => {
+    const { findByText } = render(<TestComponent />)
+
+    fireEvent.keyDown(document, { code: "KeyA" })
+    expect(await findByText("Pressed a")).toBeInTheDocument()
+  })
+
+  it("handles digit characters", async () => {
+    const { findByText } = render(<TestComponent />)
+
+    fireEvent.keyDown(document, { code: "Digit9" })
+    expect(await findByText("Pressed 9")).toBeInTheDocument()
+  })
+
+  it("handles ?", async () => {
+    const { findByText } = render(<TestComponent />)
+
+    fireEvent.keyDown(document, { shiftKey: true, code: "Minus" })
+    expect(await findByText("Pressed ?")).toBeInTheDocument()
+  })
+
+  it("handles ctrl + char", async () => {
+    const { findByText } = render(<TestComponent />)
+
+    fireEvent.keyDown(document, { ctrlKey: true, code: "KeyA" })
+    expect(await findByText("Pressed ctrl + a")).toBeInTheDocument()
+  })
+
+  it("handles ctrl + shift + char", async () => {
```

Comment:
Everything looks good! Maybe include in this tests though that "Pressed ctrl + a" is NOT included in the document. And then maybe the same for the other 'it("handles...)' tests?

Reply:
Good suggestion. I'll make sure to add at least one test to test the mature propagation